src/core/libraries/kernel/net.h

```diff
@@ -0,0 +1,58 @@
+// SPDX-FileCopyrightText: Copyright 2024 shadPS4 Emulator Project
+// SPDX-License-Identifier: GPL-2.0-or-later
+
+#pragma once
+
+#include <chrono>
+#include <sys/types.h>
+#include "common/types.h"
+
+#ifdef WIN32
+#define WIN32_LEAN_AND_MEAN
+#define _WINSOCK_DEPRECATED_NO_WARNINGS
+#include <Ws2tcpip.h>
+#include <iphlpapi.h>
+#include <winsock2.h>
+typedef SOCKET s_socket;
+#else
+#include <sys/socket.h>
+typedef int s_socket;
+#endif
+#include <map>
+#include <mutex>
+
+namespace Common {
+class NativeClock;
+}
+
+namespace Core::Loader {
+class SymbolsResolver;
+}
+
+namespace Libraries::Kernel {
+int PS4_SYSV_ABI posix_socket(int domain, int type, int protocol);
+int PS4_SYSV_ABI posix_connect(int sockfd, const struct sockaddr* addr, socklen_t addrlen);
```

Comment:
i am not sure, `struct socaddr ` is crossplarform, maybe explicit use OrbisNetSockaddr, and convert to win/mac/linux socaddr inside?
do not remember if these structs equals